### ISSUE TITLE
Redefined outer name not detected across modules when parallelization used

### ISSUE THREAD

**Issue by alangefe** [pylint-dev/pylint#8880]
### Bug description

When I enable parallelization (using the `-j 0` command line option), pylint no longer emits a `redefined-outer-name` warning. It will emit the warning if parallelization is disabled or if `-j 1` is used.

The files below demonstrate the issue. In this case, `test_demo()` defines `demo.some_redefined_variable`, but the variable is shadowed in `demo.main()`.

demo.py:

```
"""Module that redefines a name defined
in an outer scope
"""

def main():
    """This function redefines some_redefined_variable"""
    some_redefined_variable = "something else"
    print(f"{some_redefined_variable}")
```

test_demo.py:

```
"""Test module that defines a name that will be redefined
in the demo module.
"""
import demo

def test_demo():
    """Test function that defines the outer name"""
    demo.some_redefined_variable = "something"
    demo.main()
```

The two conflicting outputs are below:

```
$ pylint -j 1 demo.py test_demo.py 
************* Module demo
demo.py:7:4: W0621: Redefining name 'some_redefined_variable' from outer scope (line 8) (redefined-outer-name)

-------------------------------------------------------------------
Your code has been rated at 8.57/10 (previous run: 10.00/10, -1.43)

$ pylint -j 0 demo.py test_demo.py 

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.57/10, +1.43)
```

### Command used

```shell
pylint -j 0 demo.py test_demo.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

I expect the warning to be emitted, as with the non-parallelized (`-j 1`) case. Specifically:

```
************* Module demo
demo.py:7:4: W0621: Redefining name 'some_redefined_variable' from outer scope (line 8) (redefined-outer-name)
```

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.6
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04
